apps/next/src/pages/Onboarding/flows/ConnectLedgerFlow/screens/ConnectLedgerScreen/components/DerivedAddresses.tsx

```diff
@@ -0,0 +1,123 @@
+import {
+  IconButton,
+  OutboundIcon,
+  Skeleton,
+  Stack,
+  truncateAddress,
+  Typography,
+} from '@avalabs/k2-alpine';
+import { getAvalancheAddressLink, openNewTab } from '@core/common';
+import { TokenWithBalance } from '@avalabs/vm-module-types';
+
+import { Section, SectionRow } from '@/pages/Onboarding/components/Section';
+
+import { useNativeBalanceFetcher } from '@core/ui';
+import { useCallback, useEffect, useState } from 'react';
+
+type DerivedAddressesProps = {
+  addresses: string[];
+  chainCaipId: string;
+};
+
+type BalanceInfo = {
+  isLoading: boolean;
+  balance?: TokenWithBalance;
+};
+
+type BalanceInfoMap = Map<string, BalanceInfo>;
+
+export const DerivedAddresses = ({
+  addresses,
+  chainCaipId,
+}: DerivedAddressesProps) => {
+  const { fetchBalance } = useNativeBalanceFetcher(chainCaipId);
+
+  const [balances, setBalances] = useState<BalanceInfoMap>(
+    initialBalanceInfoMap(addresses),
```

Comment:
Consider to use state initialization callback.
It's cheaper to create a small function rather than a whole map of objects
```suggestion
    () => initialBalanceInfoMap(addresses),
```